cordova-lib/src/cordova/plugin.js

```diff
@@ -168,10 +168,19 @@ module.exports = function plugin(command, targets, opts) {
                     })
                     .then(function(dir) {
                         // Validate top-level required variables
-                        var pluginVariables = pluginInfoProvider.get(dir).getPreferences(),
-                            missingVariables = pluginVariables.filter(function (v) {
+                        var pluginVariables = pluginInfoProvider.get(dir).getPreferences();
+                        var requiredVariables = [];
+                        for(var i in pluginVariables)
+                        {
+                            var v = pluginVariables[i];
+                            // discard variables with default value
+                            if (!v.default)
```

Comment:
When I test this adding a plugin that requires variables (for example, `com.phonegap.plugins.facebookconnect`) and I don't specify any, this line throws (`v` is `null`). The logic here doesn't seem to coincide with the actual value returned by `getPreferences()`.